Common/AR/source/ArcGISArViewInterface.cpp

```diff
@@ -201,42 +201,46 @@ LocationDataSource* ArcGISArViewInterface::locationDataSource() const
 }
 
 /*!
- * \brief Sets the location data source to \a locationDataSource.
- *
- * If \a locationDataSource is \c nullptr, the tracking of the LocationDataSource is disabled.
- * If \l tryUsingArKit is \c true, the AR framework is used for the tracking, not \l LocationDataSource.
+  \brief Sets the location data source to \a locationDataSource.
+
+  If \a locationDataSource is \c nullptr, the tracking of the LocationDataSource is disabled.
+  If \l tryUsingArKit is \c true, the AR framework is used for the tracking, not \l LocationDataSource.
  */
 void ArcGISArViewInterface::setLocationDataSource(LocationDataSource* locationDataSource)
 {
   if (m_locationDataSource == locationDataSource)
     return;
 
   m_locationDataSource = locationDataSource;
+  connect(m_locationDataSource, &LocationDataSource::locationChanged, this, &ArcGISArViewInterface::setLocationInternal);
+  connect(m_locationDataSource, &LocationDataSource::headingChanged, this, &ArcGISArViewInterface::setHeadingInternal);
+  emit locationDataSourceChanged();
+}
 
-  // update connection
-  disconnect(m_locationChangedConnection);
-  disconnect(m_headingChangedConnection);
+/*!
+  \brief The tracking mode controlling how the locations generated from the location data source
```

Comment:
Use Thomas' version and add a final period:

\brief Gets the tracking mode controlling how the locations generated from the location data source.